sklearn/decomposition/_nmf.py

```diff
@@ -1320,17 +1349,84 @@ def _check_params(self, X):
                 "Number of components must be a positive integer; got (n_components=%r)"
                 % self._n_components
             )
+
+        # max_iter
         if not isinstance(self.max_iter, numbers.Integral) or self.max_iter < 0:
             raise ValueError(
                 "Maximum number of iterations must be a positive "
                 "integer; got (max_iter=%r)"
                 % self.max_iter
             )
+
+        # tol
         if not isinstance(self.tol, numbers.Number) or self.tol < 0:
             raise ValueError(
                 "Tolerance for stopping criteria must be positive; got (tol=%r)"
                 % self.tol
             )
+
+        # beta_loss
+        self._beta_loss = _beta_loss_to_float(self.beta_loss)
+
+        # solver
+        allowed_solver = ("cd", "mu")
+        if self.solver not in allowed_solver:
+            raise ValueError(
+                "Invalid solver parameter: got %r instead of one of %r"
+                % (self.solver, allowed_solver)
+            )
+        if self.solver != "mu" and self.beta_loss not in (2, "frobenius"):
+            # 'mu' is the only solver that handles other beta losses than 'frobenius'
+            raise ValueError(
+                "Invalid beta_loss parameter: solver %r does not handle beta_loss = %r"
+                % (self.solver, self.beta_loss)
+            )
+        if self.solver == "mu" and self.init == "nndsvd":
+            warnings.warn(
+                "The multiplicative update ('mu') solver cannot update "
+                "zeros present in the initialization, and so leads to "
+                "poorer results when used jointly with init='nndsvd'. "
+                "You may try init='nndsvda' or init='nndsvdar' instead.",
+                UserWarning,
+            )
+
+        # alpha and regularization are deprecated in favor of alpha_W and alpha_H
+        # TODO clean up in 1.2
+        if self.alpha != "deprecated":
+            warnings.warn(
+                "'alpha' was deprecated in version 1.0 and will be removed "
+                "in 1.2. Use 'alpha_W' and 'alpha_H' instead",
```

Comment:
```suggestion
                "`alpha` was deprecated in version 1.0 and will be removed "
                "in 1.2. Use `alpha_W` and `alpha_H` instead",
```
If I recall, we need to do that when the numpydoc validation for the estimator will be attempted.